### PR TITLE
[Feature][SpecDecode] Add DeepSeek V4 DSpark eager decoding and probabilistic verification

### PR DESCRIPTION
**Pull request by QwertyJack**
### What this PR does / why we need it?

This is PR 3/4 of the DeepSeek V4 DSpark split defined in #11126. It depends on #12004 and establishes the eager correctness baseline before graph optimization.

For stacked review, the owned change is commit `431a64b18` (`qj/dspark-02-model..qj/dspark-03-eager`). The temporary `main`-based GitHub diff also contains PRs 1-2 until their dependencies merge.

This PR:

- adds `AscendDeepSeekV4DSparkProposer` as a specialization of the generic `AscendDsparkProposer` from #11765;
- builds anchor-first inputs and per-group standard DSA cache metadata;
- performs sequential Markov sampling for greedy and probabilistic requests;
- preserves processed draft logits for logits-space rejection sampling;
- hardens padded speculative input preparation, scheduled-width masking, empty-row fallback, and request-aligned metadata copies;
- leaves generic Qwen/GLM DSpark behavior on the existing proposer.

Incremental size: about `+1915/-129`, including owned tests. This is above the RFC estimate because the eager proposer, probabilistic rejection contract, and padded-input correctness regressions must land together; moving those pieces to the model or graph PR would make neither PR independently valid.

### Does this PR introduce _any_ user-facing change?

Yes. With PRs 1-2 applied, DeepSeek V4 DSpark can run through the eager drafter by setting `speculative_config.enforce_eager=true`.

### How was this patch tested?

PR 3 focused result: `68 passed`.

Cumulative PR 1-3 unit result:

```bash
python -m pytest -q \
  tests/ut/quantization/test_modelslim_config.py \
  tests/ut/sample/a2/test_probabilistic_rejection.py \
  tests/ut/sample/test_gumbel_noise.py \
  tests/ut/sample/test_rejection_sampler.py \
  tests/ut/spec_decode/test_deepseek_v4_draft_hooks.py \
  tests/ut/spec_decode/test_deepseek_v4_dspark_model.py \
  tests/ut/spec_decode/test_dspark_dsa_metadata.py \
  tests/ut/spec_decode/test_dspark_eager_proposer.py \
  tests/ut/spec_decode/test_llm_base_proposer.py \
  tests/ut/worker/a2/test_model_runner_v1.py
```

Result: `146 passed, 1 skipped`.

Also passed:

- mypy for all PR-owned Python files;
- `ruff check` and `ruff format --check` for all changed Python files;
- `python -m py_compile` for the proposer and runner paths;
- `git diff --check qj/dspark-02-model...HEAD`.

The final real W4A8 eager service, AR/AL, and coding workload evidence will be added after the cumulative PR 4 branch is validated.

References:

- Direct dependency: #12004
- RFC and four-PR plan: #11126
- Original end-to-end prototype: #11196
- Generic Qwen/GLM DSpark foundation: #11765

- vLLM version: v0.23.0
- vLLM main: https://github.com/vllm-project/vllm/commit/e5588e49bc2642670116664a7fc4096e27adb179

### CODE REVIEW

File: vllm_ascend/worker/model_runner_v1.py [hunk 1]
```diff
@@ -1763,6 +1802,30 @@ def _copy_valid_sampled_token_count(
...
+    def _mask_sampled_token_ids_to_scheduled_width(
+        sampled_token_ids: torch.Tensor,
+        spec_decode_metadata: SpecDecodeMetadata | None,
+    ) -> torch.Tensor:
+        if spec_decode_metadata is None:
+            return sampled_token_ids
+        cu_num_draft_tokens = spec_decode_metadata.cu_num_draft_tokens
+        num_draft_tokens = torch.empty_like(cu_num_draft_tokens)
+        num_draft_tokens[:1] = cu_num_draft_tokens[:1]
+        num_draft_tokens[1:] = cu_num_draft_tokens[1:] - cu_num_draft_tokens[:-1]
```
[the remark this answers is not in the document]
Comment (QwertyJack) [reply vllm_ascend/worker/model_runner_v1.py:1815]: Not applying this suggestion. These are device tensor assignments and do not perform host conversion or call item(). This helper runs in the Python scheduling path outside the compiled model forward and ACLGraph capture. torch.diff could be evaluated separately, but there is no correctness issue or measured performance benefit that justifies changing the established path here.
